tracing/decorators.py

```diff
@@ -28,7 +28,7 @@ def _load_config(config_path: pathlib.Path) -> PyTypesToml:
     )
 
 
-def register(proj_root: pathlib.Path | None = None):
+def register(proj_root: Union[pathlib.Path, None] = None):
```

Comment:
Do not change my type hints without good reason please. 
Using | is the preferred syntax for indicating union types since the syntax was introduced
See https://peps.python.org/pep-0604/

Reply:
python3.9. does not allow using | as a type hint.

Reply:
Ah I see. Do you have python 3.10? It is the version I am using to develop. We should probably bump this in the pyproject.toml file. The CI also uses 3.10 for testing + execution

Reply:
Ah, sry, I thought we're using 3.9 since poetry uses that version if I remember correctly. Alright, I will upgrade to 3.10